client/src/js/components/bhFiscalPeriodSelect.js

```diff
@@ -33,7 +55,7 @@ function FiscalPeriodSelect(Fiscal, moment) {
     return -1;
   }
 
-  function loadPeriodsForFiscalYear(fiscalId) {
+  function loadPeriodsForFiscalYear(fiscalId, fn) {
```

Comment:
This style is a bit weird to read, since we are mixing callbacks and promises.  Generally, you think about using one or the other.

I propose you do this operation in two steps.  First, you return `Fiscal.getPeriods()` which will return a promise.  Then you chain these callbacks with a `.then()` method.  The periods will be available in the `$ctrl.periods` variable, or you can pass them back.  Like so:
```js
return Fiscal.getPeriods( /* etc */);
```
... in the `$onChanges()` function:
```js
loadPeriodsForFiscalYear($ctrl.fiscalId)
  .then(periods => {
     periods.forEach(period => {
        if (period.id === $ctrl.periodFrom) {
          // etc 
        }
     });
  });